src/code.sh

```
@@ -81,150 +53,342 @@ _trim_fastq_endings () {
+       R2_test=$(_trim_fastq_endings "_R2_" "${R2[@]}")
+
+       # Test that when "R1" and "R2" are removed the two arrays have identical file names
+       for i in "${!R1_test[@]}"; do
+              if [[ ! ${R2_test} =~ ${R1_test[$i]} ]]; then
+                     echo "Each R1 FASTQ does not appear to have a matching R2 FASTQ"
+                     echo "${R2_test} ${R1_test[$i]}"
                      exit 1
               fi
        done
+
+       _compare_fastq_name_to_prefix "$prefix" ${R1_test[@]}
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)